File: go-client-sdk/pkg/models/operations/postsubscriptionssubscriptionidunschedulependingcancellation.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/v2/pkg/models/shared"
)

type PostSubscriptionsSubscriptionIDUnschedulePendingCancellationPathParams struct {
	SubscriptionID string `pathParam:"style=simple,explode=false,name=subscription_id"`
}

type PostSubscriptionsSubscriptionIDUnschedulePendingCancellationRequest struct {
	PathParams PostSubscriptionsSubscriptionIDUnschedulePendingCancellationPathParams
}

type PostSubscriptionsSubscriptionIDUnschedulePendingCancellationResponse struct {
	ContentType  string
	StatusCode   int
	Subscription *shared.Subscription
}

File: typescript-client-sdk/src/sdk/models/operations/postsubscriptionssubscriptionidunschedulependingcancellation.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class PostSubscriptionsSubscriptionIdUnschedulePendingCancellationPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=subscription_id" })
  subscriptionId: string;
}


export class PostSubscriptionsSubscriptionIdUnschedulePendingCancellationRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: PostSubscriptionsSubscriptionIdUnschedulePendingCancellationPathParams;
}


export class PostSubscriptionsSubscriptionIdUnschedulePendingCancellationResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  subscription?: shared.Subscription;
}
